### ISSUE TITLE
st2ctl tests

### ISSUE THREAD

**Issue by armab** [StackStorm/st2tests#16]
As requested [here](https://github.com/StackStorm/st2/pull/1624#discussion_r32327755). Creating this issue as a reminder to return here later.

@Kami what could be tested here, can you give some guidance? `st2ctl reload` (with possible parameters) and `st2ctl clean`?

**Comment by Kami** [comment]
Those would be end to end tests - making sure that `start` starts all the services, `stop` stops them, you can start / restart / stop every service, etc.

**Comment by armab** [comment]
Is it possible to test that st2 could start/stop from StackStorm itself?

**Comment by Kami** [comment]
Yeah, it's possible, but we would need to use a separate instance from the one which we are testing. The instance which is being tested would need to run inside a vm / container or on a different host (build server).

**Comment by armab** [comment]
But without separated instance things I can do at minimum (test stackstorm from stackstorm itself) is only `st2ctl reload` and `st2ctl clean`, right?

**Comment by Kami** [comment]
You can test reload, but you can't really test clean since it will mess the installation up (it cleans the db).

Ideally, you should really use a separate (short lived) instance for testing.
